### ISSUE TITLE
Notifier saying published despite the fact that item was only drafted

### ISSUE THREAD

**Issue by tropcicstefan** [OrchardCMS/OrchardCore#9861]
### Describe the bug
Setting PublishContentContext.Cancel = true, doesn't change notifier message in cms accordingly. 

```    
public override Task PublishingAsync(PublishContentContext context, RadioEpisodePart instance)
{
    context.Cancel = true;
    return Task.CompletedTask;
}
```
### Expected behavior
Notifier message should indicate unsuccessful save, either by message of 'unsuccessful publish' or 'item was saved as draft'.

**Comment by tropcicstefan** [comment]
hoooh, there are some inconsistency. That Cancel flag isn't even checked on CreateAsync in DefaultContentManager

**Comment by jtkech** [comment]
Yes good catch, and there are 2 or 3 other places where it is not checked. It is only checked in the `PublishAsync()` method, but anyway this is not what would prevent the notification.

When we call  `PublishAsync()`, normally all validations have passed, so if a validation fails only the error messages are displayed. When using a controller + editor, normally validations are done by parts / fields drivers / handlers, otherwise e.g. through a REST api / workflow we now have `ValidatingAsync() / ValidatedAsync()` handlers.

I was not aware of this property., maybe it has been added at some point but looks like it is no more useful, to not be confusing, I think we could just remove this `PublishContentContext.Cancel` property.

**Comment by tropcicstefan** [comment]
> When using a controller + editor, normally validations are done by parts / fields drivers / handlers

Could you be more specific, about how to validate part through 'handler' if item is edited using a controller + editor?  Do you mean with ContentPartHandler, because ValidatingAsync() / ValidatedAsync() aren't called (only in case of REST api/workflow as you said)? I am aware of ContentPartDisplayDriver/ContentFieldDisplayDriver and adding errors to MVC ModelBinding...

> ...but looks like it is no more useful, to not be confusing, I think we could just remove this PublishContentContext.Cancel property.

my thought exactly
